src/main/java/org/opensearch/security/support/SecurityUtils.java

```diff
@@ -140,4 +149,52 @@ private static String resolveEnvVar(String envVarName, String mode, boolean bc,
             return bc ? Hasher.hash(envVarValue.toCharArray(), settings) : envVarValue;
         }
     }
+
+    // This is used to match host names - for e.g. *.example.local to dev.example.local
+    public static boolean matchesHostNamePatterns(WildcardMatcher hostMatcher, InetAddress address, String hostResolverMode) {
+        if (address == null || hostMatcher == null) {
+            return false;
+        }
+
+        // Only proceed with hostname checks if hostResolverMode is configured
+        if (hostResolverMode != null
+            && (hostResolverMode.equalsIgnoreCase("ip-hostname") || hostResolverMode.equalsIgnoreCase("ip-hostname-lookup"))) {
+            try {
+                List<String> valuesToCheck = new ArrayList<>(List.of(address.getHostAddress()));
+                final String hostName = address.getHostName();
+                valuesToCheck.add(hostName);
+                return valuesToCheck.stream().anyMatch(hostMatcher);
+            } catch (Exception e) {
+                log.warn("Failed to resolve hostname for {}: {}", address.getHostAddress(), e.getMessage());
+                return false;
+            }
+        }
+        return false;
+    }
+
+    // This is used to match IP addresses - for e.g. 192.168.0.1 to 192.168.0.0/16
+    public static boolean matchesIpAndCidrPatterns(ClientBlockRegistry<InetAddress> clientBlockRegistry, InetAddress address) {
+        if (address == null || clientBlockRegistry == null) {
+            return false;
+        }
+
+        AuthFailureListener authFailureListener = (AuthFailureListener) clientBlockRegistry;
+        final String hostAddress = address.getHostAddress();
+
+        return authFailureListener.getIgnoreHosts().stream().anyMatch(pattern -> {
+            // Handle CIDR patterns
+            if (pattern.indexOf('/') != -1) {
+                try {
+                    SubnetInfo subnetInfo = cidrCache.computeIfAbsent(pattern, authFailureListener::getSubnetForCidr);
+                    return subnetInfo.isInRange(hostAddress);
```

Comment:
If I see it correctly, `SubnetInfo` only supports IPv4. If `hostAddress` is in IPv6 format, it will throw an `IllegalArgumentException`:

https://github.com/apache/commons-net/blob/aeaa37e9753ce9920c91d22493e074781ef69ff3/src/main/java/org/apache/commons/net/util/SubnetUtils.java#L395C19-L395C43

I guess, we also need to support IPv6?

Reply:
sure, will add support for ipv6.

Reply:
Just as a recommendation: I have used in other projects this library:  

https://seancfoley.github.io/IPAddress/ 

It supports both IPv4 and IPv6, is super flexible and has a clean API.

Reply:
@nibix yes, had found this library - wasn't sure if we should use it. If it seems okay, I will try this one out.